lib/Conversion/CombToAIG/CombToAIG.cpp

```diff
@@ -428,9 +442,204 @@ struct CombAddOpConversion : OpConversionPattern<AddOp> {
       carry = rewriter.create<comb::OrOp>(op.getLoc(),
                                           ValueRange{andOp, nextCarry}, true);
     }
+    LLVM_DEBUG(llvm::dbgs() << "Lower comb.add to Ripple-Carry Adder of width "
+                            << width << "\n");
 
     rewriter.replaceOpWithNewOp<comb::ConcatOp>(op, results);
-    return success();
+  }
+
+  // Implement a parallel prefix adder - with Kogge-Stone or Brent-Kung trees
+  // Will introduce unused signals for the carry bits but these will be removed
+  // by the AIG pass.
+  void LowerParallelPrefixAdder(comb::AddOp op, ValueRange inputs,
+                                ConversionPatternRewriter &rewriter) const {
+    auto width = op.getType().getIntOrFloatBitWidth();
+
+    auto aBits = extractBits(rewriter, inputs[0]);
+    auto bBits = extractBits(rewriter, inputs[1]);
+    // Construct propagate (p) and generate (g) signals
+    SmallVector<Value> p, g;
```

Comment:
nit: Please reserve beforehand. 
```suggestion
    SmallVector<Value> p, g;
    p.reserve(width);
    g.reserve(width);
```